torch/_inductor/codegen/common.py

```diff
@@ -85,6 +85,10 @@ def propagate_node(node: torch.fx.Node):
                 opt_ctx.dtype = reduction_to_dtype[reduction_type]
         elif _node.target == "load":
             opt_ctx.dtype = V.graph.get_dtype(_node.args[1])
+        elif "masked_subblock" in _node.target:
+            assert len(masked_subblock_dtypes) != 0
+            assert _node.target in masked_subblock_dtypes
+            opt_ctx.dtype = masked_subblock_dtypes[_node.target]
```

Comment:
is it possible the called subblock has not had the dtype propagated? I guess you would have to do things recursively here?

Reply:
For the current design, it is not possible. I checked the previous CI failures and found indeed the subblock will have a subblock, but "they are all attached in the same LoopBody and ordered by dependency".
For example:
```
LoopBody:
  subblocks:
    subblock2 - no dependency
    subblock4 - dependence on subblock2
    other subblocks
  root block - dependence on subblocks
```
So we do not need recursively here.

Reply:
Firstly, it is not by design. Note that the subblocks is a dict without the guarantee of the orders. Secondly, implementing it in a recursive way is more natural for this tree-like structure.

Reply:
Hi, @jgong5. I rebased the PR and used a recursive way to handle the subblock.
By the way, I investigated why the subblocks are ordered by their dependency because they are created in a recursive way:
For example:
`subblock1` depends on `subblock 2`
while trying to create `subblock1`, we find we need to create `subblock` 2 first. And after successfully creating `subblock2` we will return to create subblock 1.
That caused the order of `subblocks` is `subblock2 -> subblock1`.